src/main/scala/nl.knaw.dans.easy.deposit/docs/DatasetMetadata.scala

```diff
@@ -67,11 +67,11 @@ case class DatasetMetadata(private val identifiers: Option[Seq[SchemedValue]] =
                                                  else Failure(missingValue("AcceptDepositAgreement"))
 
   //// dates
-  private val specialDateQualifiers = Seq(DateQualifier.created, DateQualifier.available)
+  private val specialDateQualifiers = Seq(DateQualifier.created, DateQualifier.available).map(Some(_))
```

Comment:
If you replace this whole section (https://github.com/DANS-KNAW/easy-deposit-api/blob/6646c78a249c10f5a37f14657c076c2d10d03bf6/src/main/scala/nl.knaw.dans.easy.deposit/docs/DatasetMetadata.scala#L70-L79) with the following, I think you're better off in the end. Now, you have to have `datesCreated` and `datesAvailable` and later check that it's at most one; even though your type still allows for multiple of these...

```scala
private val results = dates.getOrElse(Seq.empty)
  .foldLeft((Option.empty[Date], Option.empty[Date], Seq.empty[Date])) {
    // @formatter:off
    case ((_,           _,             _     ),      Date(_, _, Some(q@DateQualifier.dateSubmitted))) => throw new IllegalArgumentException(s"requirement failed: No $q allowed; got ${ toJson(dates) }")
    case ((None,        dateAvailable, others), date@Date(_, _, Some(  DateQualifier.created))      ) => (Some(date),  dateAvailable, others)
    case ((Some(_),     _,             _     ),      Date(_, _, Some(  DateQualifier.created))      ) => throw new IllegalArgumentException(s"requirement failed: At most one allowed; got ${ toJson(dates) }")
    case ((dateCreated, None,          others), date@Date(_, _, Some(  DateQualifier.available))    ) => (dateCreated, Some(date),    others)
    case ((_,           Some(_),       _     ),      Date(_, _, Some(  DateQualifier.available))    ) => throw new IllegalArgumentException(s"requirement failed: At most one allowed; got ${ toJson(dates) }")
    case ((dateCreated, dateAvailable, others), date@Date(_, _, _)                                  ) => (dateCreated, dateAvailable, others :+ date)
    // @formatter:on
  }
private val (_, _, plainDates) = results
val (datesCreated, datesAvailable, _) = results
val otherDates = plainDates :+ dateSubmitted()
```

Reply:
Please note that with replacing this code, you can also get rid of `Date.atMostOne` and `Date.notAllowed`

Reply:
Had to change the lines above into something like 
```
{ dm.datesAvailable.map(src => <ddm:available>{ src.value.getOrElse("") }</ddm:available>).getOrElse("") }
```
Then it still it broke a unit test as `At most one allowed` used to report just the duplicates, not all the dates. An attempt to build the string with `Seq(date,datesAvailable)` caused compilation problems.

Reply:
perhaps you're right: if you use this `DatasetMetadata` object 'on save', I guess you have to accept anything, right? Validating on multiple `dateCreated` etc. only happens 'on submit'...

Reply:
I intended to validate AtMostOnce and NotAtAll at any time, but had less verbose error messages only reporting the problematic dates.